### PR TITLE
fix create_jwt to add kid and audience similar to aws cognito

### PR DESCRIPTION
**Pull request by lalithakolla**
#MINOR#

# Changelog

* fix create_jwt to add kid and audience similar to aws cognito

### CODE REVIEW

File: moto/cognitoidp/models.py [hunk 1]
```diff
@@ -59,23 +59,25 @@ def to_json(self, extended=False):
...
             "sub": self.users[username].id,
-            "aud": client_id,
             "token_use": "id",
             "auth_time": now,
             "exp": now + expires_in,
             "email": self.users[username].username
         }
         payload.update(extra_data)
+        if audience:
+            payload['aud'] = audience
```
Comment (lalithakolla) [inline moto/cognitoidp/models.py:74]: For cognito generated tokens there is no aud for access tokens, it is there only for id token

File: moto/cognitoidp/models.py [hunk 2]
```diff
@@ -59,23 +59,25 @@ def to_json(self, extended=False):
...
 
-        return jws.sign(payload, self.json_web_key, algorithm='RS256'), expires_in
+        return jws.sign(payload, self.json_web_key, algorithm='RS256',
+                        headers={'kid': self.json_web_key['kid']}), expires_in
```
Comment (lalithakolla) [inline moto/cognitoidp/models.py:77]: kid exists in header for cognito generated tokens

**Comment by lalithakolla** [comment]
This changes can be pushed upstream too, I am working on few other changes too to push upstream

**Comment by lalithakolla** [comment]
I need this for localcluster to be working with new auth middleware

**Review by rthille — APPROVED** [review]
👍

**Pull request merged**